### PR TITLE
apps: add brick and device sets to brick allocation functions

### PR DESCRIPTION
**Pull request by phlogistonjohn**
These changes add new brick and device types to brick_allocate.go, and tries to use them to better organize some of the code within. If it seems like a lot of change for little benefit, the expectation is this approach will become more fundamental and useful in future series.

### CODE REVIEW

**Comment by phlogistonjohn** [comment]
I'm going to rebase this on top of the patches in pr #1072 because the two conflict with each other and the changes in the other pr are smaller.

**Comment by obnoxxx** [comment]
@phlogistonjohn now that #1072  is merged, this can be rebased.

File: apps/glusterfs/brick_allocate.go [hunk 1]
```diff
@@ -16,17 +16,51 @@ import (
...
+
+func NewBrickSet(s int) *BrickSet {
+	return &BrickSet{SetSize: s, Bricks: []*BrickEntry{}}
+}
+
+func (bs *BrickSet) Add(b *BrickEntry) {
+	bs.Bricks = append(bs.Bricks, b)
+}
+
+func (bs *BrickSet) Full() bool {
```
Comment (obnoxxx) [inline apps/glusterfs/brick_allocate.go]: The `Full()` methods are not used in this PR. Would we want to use them in `Add()` and prevent from over-adding? And maybe adding a (error) return value to `Add()`?

**Comment by phlogistonjohn** [comment]
Added a godbc.Require check to Add and rebased.

File: apps/glusterfs/brick_allocate.go [hunk 2]
```diff
@@ -26,6 +27,7 @@ func NewBrickSet(s int) *BrickSet {
 }
 
 func (bs *BrickSet) Add(b *BrickEntry) {
+	godbc.Require(!bs.Full())
```
Comment (obnoxxx) [inline apps/glusterfs/brick_allocate.go]: What about adding the same check for `DeviceSet.Add()`?

Comment (phlogistonjohn) [reply apps/glusterfs/brick_allocate.go]: Of course. I'm just being a blockhead.

**Comment by phlogistonjohn** [comment]
Rebased & Updated w/ an additional improvement to variable names.

**Comment by phlogistonjohn** [comment]
Commit msg updated. Var rename patch squashed.

**Pull request merged**